static/app/components/dashboards/widgetQueryFields.tsx

```diff
@@ -186,9 +191,14 @@ function WidgetQueryFields({
   };
 
   if (displayType === 'top_n') {
-    const aggregateFields = getAggregateFields();
-    const otherFields = fields.filter(field => !!!aggregateFields.includes(field));
-    const fieldValue = aggregateFields[0];
+    const fieldValue = fields
+      .slice()
+      .reverse()
+      .find(field => {
+        const fieldStr = generateFieldAsString(field);
+        return isAggregateField(fieldStr) || isAggregateEquation(fieldStr);
+      }) as QueryFieldValue;
+    const columns = fields.filter(field => field !== fieldValue);
```

Comment:
So basically we're saying that the first aggregate field from the bottom up is the y-axis? Can we do something simpler instead like:
const fieldValue = fields[-1]
const columns = fields.slice(0, fields.length -1)

Would that work? because yAxis is always required and populated for ton events?

Reply:
Yes that would work! Updated to this, tested and cleaned up some other convoluted parts of code (don't recall why I made it so complicated lol). Also fixed a bug with default query field.